src/adapter/src/catalog/storage.rs

```diff
@@ -1531,47 +1531,12 @@ impl<'a> Transaction<'a> {
         }
     }
 
-    /// Updates all clusters with ids matching the keys of `clusters` in the transaction, to the
-    /// corresponding value in `clusters`.
-    ///
-    /// Returns an error if any id in `clusters` is not found.
-    ///
-    /// NOTE: On error, there still may be some clusters updated in the transaction. It is
-    /// up to the called to either abort the transaction or commit.
-    pub(crate) fn update_clusters(
-        &mut self,
-        clusters: BTreeMap<ClusterId, catalog::Cluster>,
-    ) -> Result<(), Error> {
-        let n = self.clusters.update(|k, _v| {
-            if let Some(cluster) = clusters.get(&k.id) {
-                Some(ClusterValue {
-                    name: cluster.name().to_string(),
-                    linked_object_id: cluster.linked_object_id(),
-                    owner_id: cluster.owner_id,
-                    privileges: cluster.privileges().all_values_owned().collect(),
-                    config: cluster.config.clone(),
-                })
-            } else {
-                None
-            }
-        })?;
-        let n = usize::try_from(n).expect("Must be positive and fit in usize");
-        if n == clusters.len() {
-            Ok(())
-        } else {
-            let update_ids: BTreeSet<_> = clusters.into_keys().collect();
-            let cluster_ids: BTreeSet<_> = self.clusters.items().keys().map(|k| k.id).collect();
-            let mut unknown = update_ids.difference(&cluster_ids);
-            Err(SqlCatalogError::UnknownCluster(unknown.join(", ")).into())
-        }
-    }
-
     /// Updates cluster replica `replica_id` in the transaction to `replica`.
     ///
     /// Returns an error if `replica_id` is not found.
     ///
     /// Runtime is linear with respect to the total number of cluster replicas in the stash.
-    /// DO NOT call this function in a loop, use [`Self::update_cluster_replicas`] instead.
```

Comment:
would it make sense to keep update_cluster_replicas and delete this function?

Reply:
Maybe, the one benefit this function has is that you don't need to allocate a map. I'll mull it over during the weekend. I added `update_cluster_replicas` specifically for the migration that I'm now deleting.

Reply:
I think I'm just going to leave it as is. Re-creating this function is easy enough if anyone needs it.